### PR TITLE
Combobox fires custom onChange events when option is selected

### LINKED ISSUE #1730: [fix] ComboBox doesn't fire change events on input
**ReactUSWDS Version & USWDS Version:**

ReactUSWDS: 2.4.1
USWDS: 2.11.2 (or all)

**Describe the bug**

Changing the value of the ComboBox component does not fire the `onChange` event for the text input or the select element. If you look at the [USWDS source code](https://github.com/uswds/uswds/blob/v2.11.2/src/js/components/combo-box.js#L43) (look for the `changeElementValue` function) these events are fired when:
- mount if there is a selected option
- when an option is selected
- when the input is cleared or reset

**To Reproduce**

1. Add an input `onChange` handler prop to an instance of the ComboBox
2. This handler will be called when typing into the text input, but _not_ when an option is selected

**Expected behavior**

I would expect onChange events to be called consistently with the USWDS implementation

### PR DESCRIPTION
**Pull request by Igarfinkle**
# Summary

If a custom onChange prop is passed to the input or the select in ComboBox, that function will be called whenever a ComboBox option is selected. 

## Related Issues or PRs

Closes #1730

### CODE REVIEW

File: src/components/forms/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -399,6 +423,7 @@ export const ComboBox = forwardRef(
           onKeyDown={handleInputKeyDown}
           value={state.inputValue}
           focused={state.focusMode === FocusMode.Input}
+          inputRef={inputRef}
```
Comment (brandonlenz) [inline src/components/forms/ComboBox/ComboBox.tsx:426]: Should this be?:

```suggestion
          ref={inputRef}
```

**Comment by brandonlenz** [comment]
I should note here that while debugging some discrepancies with Isaac yesterday, I was trying to better understand the default behavior or the USDWS combo box. 

By opening the [USWDS component example](https://designsystem.digital.gov/components/combo-box/), opening the console, and running the following:

```javascript
const input = document.querySelector("#fruit");
input.addEventListener('change', (event) => {
    console.log("input changed...")
});
```

I was able to listen for `change` events on the `input` to see when it actually fired. As the github issue states, change events are only fired if the input selection, or reset happens:

https://user-images.githubusercontent.com/15805554/142665768-e0dde24e-fad6-4194-b5ef-4f7126f550ae.mov

Posting this on the ticket as a reference, mostly, as it took me a bit to confirm what the behavior should be.

File: src/components/forms/ComboBox/ComboBox.stories.tsx [hunk 2]
```diff
@@ -40,6 +40,8 @@ export const defaultComboBoxWithPropOptions = (): React.ReactElement => {
         name="input-ComboBox"
         options={fruitList}
         onChange={noop}
+        inputProps={{ onChange: (e) => console.log('input') }}
```
Comment (brandonlenz) [inline src/components/forms/ComboBox/ComboBox.stories.tsx:43]: @suzubara am I understanding correctly that we want to be able to pass an "onChange" to the input like this, and we need to not only make sure it triggers on select/reset/mount, but also override the default behavior of triggering on when the user types within the input?

I don't see where USWDS overrides the "typing" change events. Does a JSX/React created `input` element have a different default "onChange" behavior?

**Comment by Igarfinkle** [comment]
I think I'm looking for a little more concrete guidance on the intended functionality here, if you have any.

**Comment by rpdelaney** [comment]
I'm closing this as stale. Please re-open if you disagree!

**Pull request closed without merging**